yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -271,6 +283,9 @@ private[spark] class ApplicationMaster(
     rpcEnv = RpcEnv.create("sparkYarnAM", Utils.localHostName, 0, sparkConf, securityMgr)
     waitForSparkDriver()
     addAmIpFilter()
+    // If a principal and keytab have been set, use that to create new credentials for executors
+    // periodically
+    delegationTokenRenewerOption.foreach(_.scheduleLoginFromKeytab())
```

Comment:
is this one needed?  We call this above already in run() if credentials file present